File: go/problem1.go
package main

/*
 *  Problem 1 from Project Euler
 *
 *  Find the sum of all the multiples of 3 or 5 below 1000.
 */

const N = 1000

func main() {
	sum := 0
	for i := 0; i < N; i++ {
		if (i%3 == 0) || (i%5 == 0) {
			sum += i
		}
	}
	println(sum)
}

File: elixir/problem1.exs
#!/usr/bin/env elixir

# Problem 1:
# If we list all the natural numbers below 10 that are multiples of 3 or 5,
# we get 3, 5, 6 and 9. The sum of these multiples is 23.
# Find the sum of all the multiples of 3 or 5 below 1000.

defmodule Euler do
  def filter1(n, acc) when rem(n, 3) == 0, do: n + acc
  def filter1(n, acc) when rem(n, 5) == 0, do: n + acc
  def filter1(_, acc),                     do: acc
end

1..999
|> Enum.reduce(0, &Euler.filter1/2)
|> IO.puts()
